MWPW-132693: Icon block inline variant style fix

zagi25 opened the pull request:
Changes made to inline icon-block:

- Changed icon size (32 -> 40, 64 -> 56)
-  Fixed icons elongation
-  Changed Heading font size from S to XS
-  Removed width from bio inline blocks that are part of xx-up, so it can fill width of the column
- Added spacing between CTAs in action-area
- Changed font size of link in action-area to XS

After PR is approved, kitchen sink and sidekick lib should be updated (I can update the kitchen sink, but I don't have permission to edit library/block directory)

Resolves: [MWPW-132693](https://jira.corp.adobe.com/browse/MWPW-132693)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/ratko/icon-block-inline?martech=off
- After: https://icon-block-inline--milo--zagi25.hlx.page/drafts/ratko/icon-block-inline?martech=off

JasonHowellSlavin commented on libs/blocks/icon-block/icon-block.js:
```diff
@@ -39,8 +39,18 @@ const iconBlocks = {
 
+//checking if block is part of xx-up and is inline variant
+function upAndInline(el) {
+  const isInline = el.classList.contains('inline');
+  if(!isInline) return false;
+  const sectionMetadata = el.parentElement?.querySelector('.section-metadata');
+  if(!sectionMetadata) return false;
```
nit, could make a bit more concise.  Maybe something like? 

```suggestion
  const isInlineAndSectionMetadata = el.classList.contains('inline') && el.parentElement?.querySelector('.section-metadata')
  if(!isInlineAndSectionMetadata) return false;
```

zagi25 replied:
Yeah, this looks better. Thank you.

JasonHowellSlavin commented on libs/blocks/icon-block/icon-block.js:
```diff
@@ -57,7 +67,7 @@ function decorateContent(el) {
     const size = getBlockSize(el, 2);
     const variant = [...variants].filter((v) => el.classList.contains(v))?.[0] ?? 'fullwidth';
-    decorateBlockText(el, iconBlocks[size][variant]);
+    decorateBlockText(el, isUpAndInline ? ['xs', 's'] : iconBlocks[size][variant]);
```
Can we instantiate this ternary somewhere else? I don't think it being a param is particularly clear.

zagi25 replied:
`const textSize = upAndInline(el) ? ['xs', 's'] : iconBlocks[size][variant];`
`decorateBlockText(el, textSize);`
Something like this ?

JasonHowellSlavin reviewed (commented):
Good stuff. Could you add some tests for your changes, maybe also try this with `two-up` etc. on your authored pages?

rgclayton commented on libs/blocks/icon-block/icon-block.css:
```diff
@@ -103,14 +103,16 @@
+.icon-block.inline.icon-height-40 .icon-area picture img {
+  min-width:40px;
+  height: 40px;
 }
```
This change will require re-authoring of any page that uses the `icon-height-32` variant. Has this been communicated?

rgclayton replied:
I think the impact is low, but did this ever get communicated to the community?

rgclayton replied:
I posted in #milo-community relaying the breaking change.

rgclayton commented on libs/blocks/icon-block/icon-block.css:
```diff
@@ -103,14 +103,16 @@
+.icon-block.inline.icon-height-56 .icon-area,
+.icon-block.inline.icon-height-56 .icon-area picture img {
+  min-width: 56px;
+  height: 56px;
```
Same with this. Any content that is using `icon-height-64` will need re-authored with this change.

rgclayton commented on libs/blocks/icon-block/icon-block.css:
```diff
@@ -201,16 +207,16 @@
+.icon-block.inline.icon-height-40.bio .icon-area picture img {
+  height: 40px;
+  width: 40px;
 }
```
Same here.

rgclayton commented on libs/blocks/icon-block/icon-block.css:
```diff
@@ -201,16 +207,16 @@
+.icon-block.inline.icon-height-56.bio .icon-area picture img {
+  height: 56px;
+  width: 56px;
 }
```
Same here.

rgclayton reviewed (changes requested):
Icons for `inline` variants with out specific `icon-heights` are still being smooshed.

![image](https://github.com/adobecom/milo/assets/2539954/a3fc6401-9d83-4277-bfed-ccec03a2806f)

ryanmparrish commented on libs/blocks/icon-block/icon-block.css:
on line 160: 
You should include an additional `.icon-block .foreground .text-content .heading-s` selector since this changed and also needs headline bottom margin.

ryanmparrish replied:
Thanks @zagi25 
One more similar spacing issue i'm seeing on the first icon block w regards to needing bottom margin on the Icon asset. Can you extend these styles to include additional `.icon-block.full-width.small .foreground .icon-area` ex as well? 

https://github.com/adobecom/milo/pull/881/files#diff-cd28b64751555fd1edeb00e5befdd09a237354424c9880ae5621ee81a13fec68L124-L130

```
.icon-block.full-width.small .foreground .icon-area,
.icon-block.vertical.small .foreground .icon-area {
  margin-bottom: var(--spacing-s);
}

.icon-block.full-width.medium .foreground .icon-area,
.icon-block.vertical.medium .foreground .icon-area {
  margin-bottom: var(--spacing-m);
}
```

zagi25 replied:
@ryanmparrish  `var(--spacing-m)` is default bottom margin https://github.com/adobecom/milo/blob/main/libs/blocks/icon-block/icon-block.css#L68, but it has "fullwidth" class instead of "full-width". "fullwidth" is used throughout the whole file, is this correct class or should I change all occurrences of "fullwidth" to "full-width" ?

ryanmparrish commented on libs/blocks/icon-block/icon-block.css:
```diff
@@ -331,6 +338,14 @@
     max-width: 300px;
   }
 
+  .two-up .icon-block.bio.inline .foreground,
```
you could use a single selector - `[class*=-up] .icon-block.bio.inline .foreground`

ryanmparrish commented on libs/blocks/icon-block/icon-block.css:
```diff
@@ -172,10 +174,14 @@
   margin-right: var(--spacing-s);
   white-space: nowrap;
 }
+.icon-block.inline:not(.center) .foreground .text-content .action-area > a:first-child {
```
nit: extra line

ryanmparrish commented on libs/blocks/icon-block/icon-block.css:
```diff
@@ -172,10 +174,14 @@
+.icon-block.inline:not(.center) .foreground .text-content .action-area > a {
   margin-right: var(--spacing-s);
   white-space: nowrap;
 }
```
nit: extra line for linting

ryanmparrish commented on libs/blocks/icon-block/icon-block.css:
```diff
@@ -172,10 +174,14 @@
+.icon-block.inline:not(.center) .foreground .text-content .action-area > a:first-child {
+  margin-bottom: var(--spacing-xxs);
+}
 
```
Rather than targeting the <a> tags margin(s). 
Can you try adding some styles to the .action-area using flex with a gap? 
```
.icon-block .foreground .text-content .action-area {
    display: flex;
    gap: var(--spacing-s);
}
'''
If we were to approach it at the inline block level it would be good to use the 'margin-block' properties for better localization text-orientation support. Just a thought on that.

zagi25 replied:
```
  column-gap: var(--spacing-s);
  row-gap: var(--spacing-xs);
```
I used flex and this gaps instead of margins. Is this what you meant ?

zagi25 commented:
@rgclayton you are right any content using icon-height will need to be re-authored, as you saw icon-height changes were requested on the ticket. I don't think that anyone used this because it was broken ( but of course I am not 100% sure ), maybe we can leave old classes and notify authors to replace them with new ones, and then just remove the old ones ? 

I updated my code based on the comments and requests and also added somethings that I missed for five-up, I just need to write tests and also test it more manually. 

Before I push my changes I have question for you all @JasonHowellSlavin @ryanmparrish @rgclayton is there a need for this rule  https://github.com/adobecom/milo/blob/main/libs/blocks/section-metadata/section-metadata.css#L148 ? 

![Screenshot 2023-06-22 at 15 30 22](https://github.com/adobecom/milo/assets/90400759/79d5b0fe-a6ab-4796-b6e1-2b8bff05082f)

As you can see, currently this rule is breaking the five-up inline blocks (and this is for 40px icon) ? Is there a way to fix this that I am not aware of, without erasing this line ?

zagi25 commented:
@rgclayton @ryanmparrish @JasonHowellSlavin PR is updated, please take a look. Thank you.

ryanmparrish commented on libs/blocks/section-metadata/section-metadata.css:
```diff
@@ -145,7 +145,7 @@ main>.section[class*='-up']>.content {
 }
 
-.section.five-up {
+.section.five-up :not(.icon-block.inline) {
```
We probably shouldn't add block specific selectors in section-metadata.css even in the :not() case. 
Can we make this an override in the icon-block.css rather?

ryanmparrish commented on test/blocks/icon-block/icon-block.test.js:
```diff
@@ -20,15 +20,27 @@ describe('icon blocks', () => {
-        const secondColumn = inlineBlock.querySelector('.text-content .second-column');
-        expect(firstColumn).to.exist;
-        expect(secondColumn).to.exist;
+    if(block.classList.contains('inline')){
```
```suggestion
    if (block.classList.contains('inline')) {
```

Check your linting

ryanmparrish reviewed (changes requested):
This is looking good. I'd suggest we move that section-meta added style out and into the icon-block styles and double checking the icon var sizes are correct as the jira ticket was a little hard to follow. Thanks.

zagi25 commented:
@ryanmparrish @rgclayton @JasonHowellSlavin @Blainegunn  PR will be updated when I get sign off on class naming, please check [MWPW-132693](https://jira.corp.adobe.com/browse/MWPW-132693) . Thanks.

seanchoi-dev commented on libs/blocks/icon-block/icon-block.js:
```diff
@@ -39,6 +39,13 @@ const iconBlocks = {
+//checking if block is part of xx-up and is inline variant
+function upAndInline(el) {
+  const isInlineAndSectionMetadata = el.classList.contains('inline') && el.parentElement?.querySelector('.section-metadata')
+  if(!isInlineAndSectionMetadata) return false;
```
nit: our lint standard requires a space between if and (
```suggestion
  if (!isInlineAndSectionMetadata) return false;
```

ryanmparrish commented on libs/styles/variables.css:
```diff
@@ -104,6 +104,11 @@
+  /* icon sizes */
+  --l-icon: 56px;
+  --m-icon: 40px;
+
```
We should follow the same naming convention as other size vars.  elem+attr+unit
`--icon-size-m`

ryanmparrish commented on libs/blocks/icon-block/icon-block.css:
```diff
@@ -74,6 +73,16 @@
 
+.icon-block.inline .icon-area,
+.icon-block.inline .icon-area picture img {
+  min-width:80px;
```
```suggestion
  min-width: var(--xxl-icon);
```

ryanmparrish commented on libs/styles/variables.css:
```diff
@@ -104,6 +104,11 @@
   --grid-margins-width-6: calc((100vw - (var(--grid-column-width) * 6)) / 2);
+
+  /* icon sizes */
+  --l-icon: 56px;
```
```suggestion
  --xxl-icon: 80px;
  --xl-icon: 64px;
  --l-icon: 56px;
```

ryanmparrish commented on libs/styles/variables.css:
```diff
@@ -104,6 +104,11 @@
+
+  /* icon sizes */
+  --l-icon: 56px;
+  --m-icon: 40px;
```
```suggestion
  --m-icon: 40px;
  --s-icon: 32px;
  --xs-icon: 24px;
  --xxs-icon: 16px;
```

ryanmparrish commented on libs/blocks/icon-block/icon-block.css:
```diff
@@ -74,6 +73,16 @@
+
+.icon-block.inline.bio .icon-area,
+.icon-block.inline.bio .icon-area picture img {
+  min-width:60px;
```
```suggestion
  min-width: var(--xl-icon);
```

ryanmparrish replied:
Technically the xl-icon is 64px but maybe we need that re-alignment on consistency anyways

ryanmparrish commented on libs/blocks/icon-block/icon-block.css:
```diff
@@ -103,14 +112,16 @@
   width: 60px;
```
```suggestion
  width: var(--xl-icon);
```

ryanmparrish commented on libs/blocks/icon-block/icon-block.css:
```diff
@@ -103,14 +112,16 @@
-  height: 64px;
+.icon-block.inline.l-icon .icon-area,
+.icon-block.inline.l-icon .icon-area picture img {
+  min-width: var(--l-icon);
```
```suggestion
  min-width: var(--xl-icon);
```

ryanmparrish commented on libs/blocks/icon-block/icon-block.css:
```diff
@@ -103,14 +112,16 @@
+.icon-block.inline.l-icon .icon-area,
+.icon-block.inline.l-icon .icon-area picture img {
+  min-width: var(--l-icon);
+  height: var(--l-icon);
```
```suggestion
  height: var(--xl-icon);
```

ryanmparrish commented on libs/blocks/icon-block/icon-block.css:
```diff
@@ -144,6 +155,27 @@
+.five-up .icon-block.inline .foreground .icon-area,
+.five-up .icon-block.inline .foreground .icon-area img {
+  max-height: unset;
+  max-width: 80px;
```
```suggestion
  max-width: var(--xxl-icon);
```

ryanmparrish commented on libs/blocks/icon-block/icon-block.css:
```diff
@@ -267,11 +311,25 @@
     height: 80px;
```
```suggestion
    height: var(--xxl-icon)
```

ryanmparrish commented on libs/blocks/icon-block/icon-block.css:
```diff
@@ -267,11 +311,25 @@
 
+  .icon-block.inline.bio .icon-area,
+  .icon-block.inline.bio .icon-area picture img {
+    min-width:80px;
```
```suggestion
    min-width: var(--xxl-icon);
```

ryanmparrish commented on libs/blocks/icon-block/icon-block.css:
```diff
@@ -295,6 +353,33 @@
     height: 80px;
```
```suggestion
    height: var(--xxl-icon);
```

zagi25 commented:
@ryanmparrish main ticket (https://jira.corp.adobe.com/browse/MWPW-129909) was moved to 2nd review . I notified them that I will be on PTO next week and told them that if this is urgent that they should ask somebody from my team to takeover. If not I will address your comments when I return from PTO. Thank you.

zagi25 commented:
@ryanmparrish @seanchoi-dev @rgclayton @Blainegunn PR is updated, please check it out. Thanks.

ryanmparrish commented:
@zagi25 This is looking good, Thanks for those updates. I left one tiny comment in the css. 
Do you have your eslint package installed? I'm seeing a few linting errors that we should try to clean up.

seanchoi-dev commented on libs/blocks/icon-block/icon-block.js:
```diff
@@ -39,6 +39,13 @@ const iconBlocks = {
 
+//checking if block is part of xx-up and is inline variant
+function upAndInline(el) {
+  const isInlineAndSectionMetadata = el.classList.contains('inline') && el.parentElement?.querySelector('.section-metadata')
+  if (!isInlineAndSectionMetadata) return false;
+  return /(two|three|four|five)[- ]?up/i.test(el.parentElement.querySelector('.section-metadata').textContent);
```
nit: missing semicolon and simplifying the test object.
```suggestion
  const isInlineAndSectionMetadata = el.classList.contains('inline') && el.parentElement?.querySelector('.section-metadata');
  if (!isInlineAndSectionMetadata) return false;
  return /(two|three|four|five)[- ]?up/i.test(el.parentElement.className);
  ```

zagi25 replied:
Thanks, this looks better.

seanchoi-dev reviewed (approved):
Looks good to me. I commented with a small suggestion, but approving it.

rgclayton commented:
Looking much better! Thank you @zagi25 .

Some clarification questions related to icon sizing. 
New icon size variants:

1. `m-icon` (40px)
2. `l-icon` (56px)
3.  `default` (80px)

These variants only work with the `icon-block.inline` variation, correct? 

cc: @Blainegunn @ryanmparrish

rgclayton reviewed (approved):
Thanks for the updates @zagi25 

I think we are good once the requests from @ryanmparrish are added.

zagi25 commented:
PR is updated. I added css rule that @ryanmparrish requested and fixed lint errors. Thank you all for your reviews and feedback.
cc: @rgclayton @seanchoi-dev @Blainegunn @JasonHowellSlavin
